bindings/nodejs/src/functions/gpio.cc

```diff
@@ -0,0 +1,145 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <sol-gpio.h>
+
+#include <v8.h>
+#include <node.h>
+#include <nan.h>
+#include <map>
+
+#include "../common.h"
+#include "../hijack.h"
+#include "../structures/sol-js-gpio.h"
+#include "../structures/js-handle.h"
+
+using namespace v8;
+
+class SolGpio : public JSHandle<SolGpio> {
+public:
+    static const char *jsClassName() { return "SolGpio"; }
+};
+
+static void sol_gpio_read_callback(void *data, struct sol_gpio *gpio, bool value) {
+    Nan::HandleScope scope;
+    sol_gpio_data *gpio_data = (sol_gpio_data *)data;
+    Nan::Callback *callback = gpio_data->callback;
+    if (!callback)
+        return;
+
+    Local<Value> arguments[2] = {
+        SolGpio::New(gpio_data),
```

Comment:
Conceptually, the parameter ```sol_gpio *gpio``` is given as a convenience to this callback. It's not a ```sol_gpio *``` newly created soletta. It's the same pointer that you have received from soletta when you earlier called ```sol_gpio_open()```. Therefore, you should not create a second JS object to hold it, because you've already created a JS object in ```bind_sol_gpio_open()```.